internal_decision_state_machine.go

```diff
@@ -676,6 +684,28 @@ func (h *decisionsHelper) getActivityID(event *s.HistoryEvent) string {
 	return activityID
 }
 
+func (h *decisionsHelper) recordVersionMarker(changeID string, version Version) decisionStateMachine {
+	markerID := fmt.Sprintf("%v_%v", versionMarkerName, changeID)
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	if err := enc.Encode(changeID); err != nil {
```

Comment:
Why are you not going through encoding interface exposed through hostEnv.Encoder()? that would give us single place to switch different encoders.